src/vm/arm/ehhelpers.S

```diff
@@ -99,6 +99,7 @@ OFFSET_OF_FRAME=(4 + SIZEOF__GSCookie)
         NESTED_ENTRY CallEHFunclet, _TEXT, NoHandler
 
         PROLOG_PUSH  "{r4-r11, lr}"
+        PROLOG_STACK_SAVE r7
         alloc_stack  4
 
```

Comment:
.pad should have pointed r7 saved in the stack (need to add 12) or .setfp along with setting r7, pointing at r7 saved in stack is required.